### PR TITLE
Feature/set associations

### PR DESCRIPTION
**Pull request by daudi13**
In this PR I:

- Created a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for @label (remember about attr_accessor).
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Created the `has-many` side (a classroom has many students).
  - Created the belongs-to side (a student belongs to a classroom).
  - Made sure that when adding a student to a classroom it also set the classroom for the student.
  - Made sure that when setting the classroom for a student it also added it to the `classroom`  `students`.
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables, were initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Created a class `Rental` with the following:
  - `@date` instance variable, was initialized in the constructor.
  - `Setter` and `getter` for `@date` (remember about attr_accessor).
- Created the `many-to-many` (also has-many-through) relationship between `Person` and `Book` using the intermediate class `Rental`. The following were implemented:
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to side` of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  attr_accessor :title, :author
+  attr_reader :rental
+
+  def initialize(title, author)
+    super()
+    @title = title
+    @author = author
+    @rental = []
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline book.rb]: - [x] There should be a method in your book class ` add_rental` which takes `person` and `date` as arguments, this method should add the book to the rental class

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @daudi13  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (daudi13) [reply book.rb]: Thank you for pointing that out. let me work on it.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    super()
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(rental)
+    @rental.push(rental)
+    rental.book = self
+  end
```
Comment (t-yanick) [inline book.rb]: - [ ] As requested by the previous code reviewer, I think the `add_rental` method should be something like so:
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```
The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association, please kindly fix that.

File: person.rb [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
-    @id = SecureRandom.uuid
+    @id = Random.rand(1..1000)
     @age = age
     @name = name
     @parent_permission = parent_permission
+    @rental = []
   end
```
Comment (t-yanick) [inline person.rb:16]: - [OPTIONAL] To implement the `has-many` side of the association between `person` and `rental`, `a method for adding rentals should be present` in the `Person class`. Something like so;
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```
The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association, please kindly fix  that.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @daudi13, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 2 above)
Comment (daudi13) [reply book.rb]: Thanks for the feedback

**Review by Meltrust — APPROVED** [review]
Hi @daudi13,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for refactoring your `add_rental` method✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**